### PR TITLE
TS migration function generation

### PR DESCRIPTION
**Pull request by phiSgr**
python coming next

### CODE REVIEW

File: apps/framework-cli/src/framework/python/executor.rs [hunk 1]
```diff
@@ -42,6 +42,16 @@ pub static AGGREGATIONS_RUNNER: &str = include_str!("scripts/aggregations_runner
 pub static BLOCKS_RUNNER: &str = include_str!("scripts/blocks_runner.py");
 pub static CONSUMPTION_RUNNER: &str = include_str!("scripts/consumption_runner.py");
 
+const PYTHON_PATH: &str = "PYTHONPATH";
+fn python_path_with_version() -> String {
+    let mut paths = std::env::var(PYTHON_PATH).unwrap_or_else(|_| String::from(""));
+    if !paths.is_empty() {
+        paths.push(':');
+    }
+    paths.push_str(".moose/versions");
```
Comment (callicles) [inline apps/framework-cli/src/framework/python/executor.rs]: That probably should refer to a constant right?

File: apps/framework-cli/src/framework/data_model/parser.rs [hunk 2]
```diff
@@ -51,12 +54,22 @@ pub fn parse_data_model_file(
                     path, project, version,
                 )?),
                 Some("py") => Ok(python::parser::extract_data_model_from_file(path, version)?),
-                _ => Err(DataModelParsingError::UnsupportedFileType),
+                Some("pyc") => Ok(FileObjects::default()), // __pycache__
```
Comment (callicles) [inline apps/framework-cli/src/framework/data_model/parser.rs]: should those refer to constants

File: apps/framework-cli/src/framework/languages.rs [hunk 3]
```diff
@@ -9,6 +9,15 @@ pub enum SupportedLanguages {
     Python,
 }
 
+impl SupportedLanguages {
+    pub fn extension(self) -> &'static str {
+        match self {
+            SupportedLanguages::Typescript => "ts",
```
Comment (callicles) [inline apps/framework-cli/src/framework/languages.rs]: Like this to a constant

File: apps/framework-cli/src/cli/local_webserver.rs [hunk 4]
```diff
@@ -777,7 +778,13 @@ async fn management_router<I: InfraMapProvider>(
     infra_map: I,
     req: Request<Incoming>,
 ) -> Result<Response<Full<Bytes>>, hyper::http::Error> {
-    debug!(
+    let level = if req.uri().path().ends_with("metrics-logs") {
```
Comment (callicles) [inline apps/framework-cli/src/cli/local_webserver.rs]: constant?

File: apps/framework-cli/src/cli/routines/block.rs [hunk 5]
```diff
@@ -1,24 +1,23 @@
...
-
 pub async fn create_block_file(
     project: &Project,
     filename: String,
 ) -> Result<RoutineSuccess, RoutineFailure> {
     let blocks_dir = project.blocks_dir();
     let (extension, template) = match project.language {
         SupportedLanguages::Typescript => ("ts", TS_BASE_BLOCK_TEMPLATE),
-        SupportedLanguages::Python => ("py", PTYHON_BASE_BLOCKS_TEMPLATE),
+        SupportedLanguages::Python => ("py", PYTHON_BASE_BLOCKS_TEMPLATE),
```
Comment (callicles) [inline apps/framework-cli/src/cli/routines/block.rs]: constants?

File: apps/framework-cli/src/framework/python/executor.rs (hunk 1 above)
Comment (phiSgr) [inline apps/framework-cli/src/framework/python/executor.rs]: will have to add it to `python.analysis.extraPaths` `python.autoComplete.extraPaths` in vs code probably

File: apps/framework-cli/src/framework/core/primitive_map.rs [hunk 6]
```diff
@@ -86,7 +86,11 @@ impl PrimitiveMap {
             get_all_current_streaming_functions(project, &primitive_map.datamodels)
                 .await?
                 .iter()
-                .filter(|func| func.executable.extension().unwrap() == "ts")
+                .filter(|func| {
+                    func.executable
+                        .extension()
+                        .is_some_and(|ext| ext == "ts" || ext == "py")
```
Comment (callicles) [inline apps/framework-cli/src/framework/core/primitive_map.rs]: constants?

File: apps/framework-cli/src/framework/python/templates.rs [hunk 7]
```diff
@@ -88,7 +88,7 @@ def run(client, params):
     )
 "#;
 
-pub static PTYHON_BASE_BLOCKS_TEMPLATE: &str = r#"
+pub static PYTHON_BASE_BLOCKS_TEMPLATE: &str = r#"
 from dataclasses import dataclass
 from typing import List
```
Comment (callicles) [inline apps/framework-cli/src/framework/python/templates.rs:93]: What about this? does this need to be updated?

File: apps/framework-cli/src/framework/python/templates.rs [hunk 8]
```diff
@@ -110,7 +110,7 @@ setup_queries = []
 block = Blocks(teardown=teardown_queries, setup=setup_queries)
 "#;
 
-pub static PTYHON_BASE_BLOCKS_SAMPLE_TEMPLATE: &str = r#"
+pub static PYTHON_BASE_BLOCKS_SAMPLE: &str = r#"
 from dataclasses import dataclass
 from typing import List
```
Comment (callicles) [inline apps/framework-cli/src/framework/python/templates.rs:115]: And this?

**Review by callicles — APPROVED** [review]
Small things to improve you could merge as is if you wanted

File: apps/framework-cli/src/framework/python/templates.rs (hunk 7 above)
Comment (phiSgr) [reply apps/framework-cli/src/framework/python/templates.rs:93]: it's a typo fix :P

File: apps/framework-cli/src/framework/python/templates.rs (hunk 8 above)
Comment (phiSgr) [reply apps/framework-cli/src/framework/python/templates.rs:115]: it's not a template :/

File: apps/framework-cli/src/framework/data_model/parser.rs [hunk 9]
```diff
@@ -50,11 +50,13 @@ pub fn parse_data_model_file(
...
-                )?),
-                Some("py") => Ok(python::parser::extract_data_model_from_file(path, version)?),
-                Some("pyc") => Ok(FileObjects::default()), // __pycache__
+                Some(constants::TYPESCRIPT_FILE_EXTENSION) => Ok(
+                    typescript::parser::extract_data_model_from_file(path, project, version)?,
+                ),
+                Some(constants::PYTHON_FILE_EXTENSION) => {
+                    Ok(python::parser::extract_data_model_from_file(path, version)?)
+                }
+                Some(constants::PYTHON_CACHE_EXTENSION) => Ok(FileObjects::default()), // __pycache__
```
Comment (phiSgr) [inline apps/framework-cli/src/framework/data_model/parser.rs:59]: is it clearer than the string literals? 🥴

**Pull request merged**